synapse/rest/client/profile.py

```diff
@@ -27,6 +28,24 @@
     from synapse.server import HomeServer
 
 
+def _read_propagate(hs: "HomeServer", request: SynapseRequest) -> bool:
+    # This will always be set by the time Twisted calls us.
+    assert request.args is not None
+
+    propagate = True
+    if hs.config.experimental.msc4069_profile_inhibit_propagation:
+        do_propagate = request.args.get(b"org.matrix.msc4069.propagate")
+        if do_propagate is not None:
+            do_propagate = do_propagate[0].lower()
+            if do_propagate not in [b"true", b"false"]:
+                raise SynapseError(
+                    400, "'propagate' is malformed", errcode=Codes.INVALID_PARAM
+                )
+            else:
+                propagate = do_propagate == b"true"
```

Comment:
```suggestion
        do_propagate = request.args.get(b"org.matrix.msc4069.propagate")
        if do_propagate is not None:
            propogate = parse_boolean(
                request, "org.matrix.msc4069.propagate", default=False
            )
```

I think should do it (with `from synapse.http.servlet import parse_boolean`)

Reply:
Applied manually